modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/test/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelperTest.java

```diff
@@ -258,6 +261,63 @@ public void testBelongsToCondition() throws Exception {
 		Assert.assertTrue((boolean)ddmFormFieldPropertyChanges.get("readOnly"));
 	}
 
+	@Test
+	public void testDDMFormPageChange() throws Exception {
```

Comment:
@carolmariaabb, can you please give me more details about the purpose of this test? What exactly is it trying to check / validate? :thinking: 

Will this test check the same scenario being covered by the integration test that @brunofariasdeo will work soon? Do we need both tests or the integration test should be enough?

Reply:
Sure, @natocesarrego!
This test validates if the field properties are changing correctly according to the implementation of the `DDMFormPageChange`. For example, [here](https://github.com/carolmariaabb/liferay-portal/commit/380f627e0983af9993e6f06690c01ab76c4d27dc#diff-09119d3022efd7e70459a6a00be82d1ff216f7c400c1a8ea20e00f66e15af36fR1853) I changed the `value` property of `field1` and this [line](https://github.com/carolmariaabb/liferay-portal/commit/380f627e0983af9993e6f06690c01ab76c4d27dc#diff-09119d3022efd7e70459a6a00be82d1ff216f7c400c1a8ea20e00f66e15af36fR314) validates it.

I also added some checks to make sure the "normal" evaluation is working correctly when there is an extension point.
For example, [when a field is required and the value is empty](https://github.com/carolmariaabb/liferay-portal/commit/380f627e0983af9993e6f06690c01ab76c4d27dc#diff-09119d3022efd7e70459a6a00be82d1ff216f7c400c1a8ea20e00f66e15af36fR304).

> Will this test check the same scenario being covered by the integration test that @brunofariasdeo will work soon? 

No, because this test [guarantees that the extension point will be executed when changing the page](https://github.com/carolmariaabb/liferay-portal/commit/380f627e0983af9993e6f06690c01ab76c4d27dc#diff-09119d3022efd7e70459a6a00be82d1ff216f7c400c1a8ea20e00f66e15af36fR277), and I believe that we were unable to do this in the integration test. What do you think?

Reply:
Makes sense, @carolmariaabb, after all I don't see a way to trigger a page change from the back-end, because this is an action being performed on the front-end. Thanks for your explanation! :slightly_smiling_face: